lib/matplotlib/backends/backend_agg.py

```diff
@@ -508,8 +509,16 @@ def print_raw(self, filename_or_obj, *args, **kwargs):
         else:
             fileobj = filename_or_obj
             close = False
+        img = renderer._renderer.buffer_rgba()
+
+        # Flatten RGBA if used with fileformat that doesn't handle trnasparency
+        if kwargs.get('flatten', False):
+            w, h = int(renderer.width), int(renderer.height)
+            img = np.array(memoryview(img)).reshape((w, h, 4))
```

Comment:
Can someone verify that this is the correct reshape?  For png I needed to use (h, w, 4) to avoid scanning line offsets.  The images look correct.